### PR TITLE
Added to msg_sender to prelude and removed from swayfiles

### LINKED ISSUE #4478: Add `msg_sender()` into the prelude
It's used often, worth adding into the prelude.

### PR DESCRIPTION
**Pull request by calldelegation**
## Description
Fixes #4478

## Checklist

- [x] I have linked to any relevant issues.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

File: docs/reference/src/code/examples/access-control/ownership/src/main.sw [hunk 1]
```diff
@@ -1,8 +1,5 @@
 contract;
 
-// ANCHOR: identity
```
Comment (kayagokalp) [inline docs/reference/src/code/examples/access-control/ownership/src/main.sw:3]: Should we keep the anchor? Looks like we are still using it

**Comment by calldelegation** [comment]
Missing LSP modifications to support these changes
[Similar PR by Braqzen](https://github.com/FuelLabs/sway/pull/3916/files#diff-54dcd495bcabacd52ae149a0d941f40561fb7e9bdec3878ed66377ec113f7183)

File: docs/reference/src/code/examples/access-control/ownership/src/main.sw (hunk 1 above)
Comment (calldelegation) [reply docs/reference/src/code/examples/access-control/ownership/src/main.sw:3]: Added back in! @kayagokalp

**Comment by Braqzen** [comment]
I suggest doing a global search for `auth::` as that will come up with additional imports to remove

**Comment by calldelegation** [comment]
> I suggest doing a global search for `auth::` as that will come up with additional imports to remove

Good call I think I got them all now! 😄

**Review by Braqzen — COMMENTED** [review]
[Here](https://github.com/FuelLabs/sway/blob/add/msg_sender-to-prelude/test/src/e2e_vm_tests/test_programs/should_fail/supertraits_for_abis_ownable1/src/ownable.sw#L28) is another.

It's fine to clean up files within the related area so if you search for `auth::*` you'll find a bunch of imports too which seem to be entirely unused at this point. I would consider which file they are in and if they still serve some demonstration purpose - if not I would remove them.
One common usage is to still import the `AuthError` and use it as an explicit type hint for the user; however, unless that's used in the book (i.e. just in regular code) it's likely that it's better to omit it.

File: docs/reference/src/code/operations/call_data/src/lib.sw
Comment (Braqzen) [inline docs/reference/src/code/operations/call_data/src/lib.sw:1]: This change is related to the following [doc](https://github.com/FuelLabs/sway/blob/master/docs/reference/src/documentation/operations/call-data/msg-sender.md). Could this doc be updated to remove the comment stating that we must import and then also remove the following code block which imports?

**Review by Braqzen — COMMENTED** [review]
After the following suggestion we can likely merge

File: docs/reference/src/code/operations/call_data/src/lib.sw
Comment (calldelegation) [reply docs/reference/src/code/operations/call_data/src/lib.sw:1]: Also updated the [prelude doc here](https://github.com/FuelLabs/sway/blob/master/docs/reference/src/documentation/misc/prelude.md) to match that of this one [here](https://github.com/FuelLabs/sway/blob/master/docs/reference/src/documentation/misc/prelude.md)

**Review by Braqzen — APPROVED** [review]
That's cool. I haven't really checked the reference since it was written. It's probably quite out of date and it needs more work.

**Review by kayagokalp — APPROVED** [review]

**Pull request merged**